src/Microsoft.PowerShell.Commands.Management/commands/management/CombinePathCommand.cs

```diff
@@ -128,6 +141,18 @@ protected override void ProcessRecord()
                     continue;
                 }
 
+                // Change extension if specified
+                if (Extension != null && joinedPath != null)
+                {
+                    joinedPath = System.IO.Path.ChangeExtension(joinedPath, Extension);
+
+                    // Remove trailing dot when extension is empty string
+                    if (Extension.Length == 0)
+                    {
+                        joinedPath = joinedPath.TrimEnd('.');
+                    }
```

Comment:
@mklement0 What is your opinion about this?

@yotsuda Have we test for the code path?

Reply:
`-Extension [NullString]::Value` effectively ignore the parameter.

Reply:
To spell it out:

Inside the `if (Extension != null && joinedPath != null)` branch, the only code should be:

```c#
joinedPath = System.IO.Path.ChangeExtension(joinedPath, Extension.Length == 0 ? null : Extension);
```

Plus, the `Extension` parameter property must be declared not to accept `null`, though I'm confused about that: I thought that is the case _by default_ and that you must conversely _opt in_ to support `null`, via `[AllowNull()]`.  However, in quick tests it seems that `null` is accepted by default. So perhaps a `[ValidateNotNull]` attribute is needed too.